### ISSUE TITLE
.rcl files are inconsistent for agents and managers

### ISSUE THREAD

**Issue by RamiusKahn** [wazuh/wazuh#3740]
| Wazuh version | Component | Install type | Install method | Platform |
| --- | --- | --- | --- | --- |
| 3.9.3 | wazuh-modulesd:aws-s3 | Manager & Agent | Packages | Ubuntu18 |

1. The contents of the following files are different:
`/var/ossec/etc/shared/default/cis_win2012r2_memberL2_rcl.txt`
`/var/ossec/ruleset/sca/cis_win2012r2_memberL2_rcl.yml`

2. Content may be missing from the files. 
`/var/ossec/ruleset/sca/cis_win2012r2_memberL2_rcl.yml` on the Manager contains `638` lines  `C:\Program Files (x86)\ossec-agent\shared\cis_win2012r2_memberL2_rcl.txt` file on the agent contains:  `4731` lines.

3. The contents of `C:\Program Files (x86)\ossec-agent\shared\cis_win2012r2_memberL2_rcl.txt` and `/var/ossec/etc/shared/default/cis_win2012r2_memberL2_rcl.txt` contain multiple instances of the following string that prevent the configuration from loading on agents:

```
#
#
 -> !1;
#
#
```
3. The number and content (see file sizes for same files across both directories) for SCA folders on default installs do not match and require manual synch between SCA and shared for agents. Based on observation 2, checks may be lost or omitted by just synching the files.

```
root@elm-master:/home/ubuntu# ls -ll /var/ossec/ruleset/sca/ | grep cis
-rw-r----- 1 root ossec  66906 Jul  9 11:00 cis_apache2224_rcl.yml
-rw-r----- 1 root ossec  35113 Jul  9 11:00 cis_apple_macOS_10.11.yml
-rw-r----- 1 root ossec  32119 Jul  9 11:00 cis_apple_macOS_10.12.yml
-rw-r----- 1 root ossec  30585 Jul  9 11:00 cis_apple_macOS_10.13.yml
-rw-r----- 1 root ossec  35138 Jul  9 11:00 cis_debian_linux_rcl.yml
-rw-r----- 1 root ossec  98288 Jul  9 11:00 cis_debianlinux7-8_L1_rcl.yml
-rw-r----- 1 root ossec  37087 Jul  9 11:00 cis_debianlinux7-8_L2_rcl.yml
-rw-r----- 1 root ossec  20291 Jul  9 11:00 cis_mysql5-6_community_rcl.yml
-rw-r----- 1 root ossec  26326 Jul  9 11:00 cis_mysql5-6_enterprise_rcl.yml
-rw-r----- 1 root ossec  52837 Jul  9 11:00 cis_rhel5_linux_rcl.yml
-rw-r----- 1 root ossec  50743 Jul  9 11:00 cis_rhel6_linux_rcl.yml
-rw-r----- 1 root ossec  60289 Jul  9 11:00 cis_rhel7_linux_rcl.yml
-rw-r----- 1 root ossec  56114 Jul  9 11:00 cis_sles11_linux_rcl.yml
-rw-r----- 1 root ossec  59277 Jul  9 11:00 cis_sles12_linux_rcl.yml
-rw-r----- 1 root ossec  68191 Jul  9 11:00 cis_solaris11_rcl.yml
-rw-r----- 1 root ossec 342528 Jul  9 11:00 cis_win10_enterprise_L1_rcl.yml
-rw-r----- 1 root ossec 126036 Jul  9 11:00 cis_win10_enterprise_L2_rcl.yml
-rw-r----- 1 root ossec 258521 Jul  9 11:00 cis_win2012r2_domainL1_rcl.yml
-rw-r----- 1 root ossec  71884 Jul  9 11:00 cis_win2012r2_domainL2_rcl.yml
-rw-r----- 1 root ossec 281994 Jul  9 11:00 cis_win2012r2_memberL1_rcl.yml
-rw-r----- 1 root ossec  52703 Jul  9 11:00 cis_win2012r2_memberL2_rcl.yml
root@elm-master:/home/ubuntu# ls -ll /var/ossec/etc/shared/default/ | grep cis
-rw-rw---- 1 ossec  ossec  28416 Jun  7 14:11 cis_apache2224_rcl.txt
-rw-rw---- 1 ossec  ossec  12581 Jun  7 14:11 cis_debian_linux_rcl.txt
-rw-rw---- 1 ossec  ossec   7614 Jun  7 14:11 cis_mysql5-6_community_rcl.txt
-rw-rw---- 1 ossec  ossec  10302 Jun  7 14:11 cis_mysql5-6_enterprise_rcl.txt
-rw-rw---- 1 ossec  ossec  35786 Jun  7 14:11 cis_rhel5_linux_rcl.txt
-rw-rw---- 1 ossec  ossec  33875 Jun  7 14:11 cis_rhel6_linux_rcl.txt
-rw-rw---- 1 ossec  ossec  36962 Jun  7 14:11 cis_rhel7_linux_rcl.txt
-rw-rw---- 1 ossec  ossec  17663 Jun  7 14:11 cis_rhel_linux_rcl.txt
-rw-rw---- 1 ossec  ossec  34381 Jun  7 14:11 cis_sles11_linux_rcl.txt
-rw-rw---- 1 ossec  ossec  35086 Jun  7 14:11 cis_sles12_linux_rcl.txt
-rw-rw---- 1 ossec  ossec  94882 Jun  7 14:11 cis_win2012r2_domainL1_rcl.txt
-rw-rw---- 1 ossec  ossec  28011 Jun  7 14:11 cis_win2012r2_domainL2_rcl.txt
-rw-rw---- 1 ossec  ossec 100535 Jun  7 14:11 cis_win2012r2_memberL1_rcl.txt
-rw-rw---- 1 ossec  ossec 376007 Jun  7 14:11 cis_win2012r2_memberL2_rcl.txt

```

**Comment by Zenidd** [comment]
Hi @RamiusKahn,

> The contents of the following files are different:
/var/ossec/etc/shared/default/cis_win2012r2_memberL2_rcl.txt
/var/ossec/ruleset/sca/cis_win2012r2_memberL2_rcl.yml

These two files are used by two different Wazuh components. Both contain policies for the same OS version and they share some checks. 

- .txt policy files: [`Rootcheck`](https://documentation.wazuh.com/3.x/user-manual/capabilities/policy-monitoring/rootcheck/rootcheck-configuration.html)
- .yml policy files: [`SCA`](https://documentation.wazuh.com/3.x/user-manual/capabilities/sec-config-assessment/security-configuration-assessment.html)

We don't expect them to have exactly the same check list.

> Content may be missing from the files.
/var/ossec/ruleset/sca/cis_win2012r2_memberL2_rcl.yml on the Manager contains 638 lines C:\Program Files (x86)\ossec-agent\shared\cis_win2012r2_memberL2_rcl.txt file on the agent contains: 4731 lines.

`SCA`(.yml files) targets compliant requirements while `Rootcheck`(.txt files) contains checks beyond of the compliant requirements.

`SCA` is included in Wazuh since v3.9. It's a relatively new component and is our priority when it comes to configuration assessment scans. We are also working on a important refactoring of this module (you can track it on https://github.com/wazuh/wazuh/issues/3249) as well as new policies which be available in Wazuh v3.10. These new policies have a different syntax than the ones used until now and are more extensive than the current ones.  [Here](https://github.com/wazuh/wazuh-ruleset/tree/404-adapt-sca-rules-to-new-syntax/sca) you can take a look at all of them. Some policy files are also deleted and unified with others.

> The number and content (see file sizes for same files across both directories) for `SCA` folders on default installs do not match and require manual synch between `SCA` and shared for agents. Based on observation 2, checks may be lost or omitted by just synching the files.

`Rootcheck` policy files are placed on `/var/ossec/etc/shared/default/` on install. Currently `SCA` policies are NOT copied on install to `/var/ossec/etc/shared/default/`. As you said, the user is expected to manually synch between `SCA` and shared for agents.


Please, let me know if it helps.

Greetings, Juan Pablo Sáez

**Comment by RamiusKahn** [comment]
Thank you for an excellent and thoughtful answer.
